### ISSUE TITLE
Field annotation produces code without namespace name

### ISSUE THREAD

**Issue by pankleks** [Microsoft/TypeScript#5760]
Hi, I have function for annotation:

```javascript
namespace planck {
  export function Watch(fn: () => void) {
      return function (target, key) { ... }
  }
}

namespace demo {
  class Test {
    @planck.Watch(function () { ... })
    Name;
  }
}
```

It compiles without any errors, but generates code without namespace "planck":

```javascript
            __decorate([
                Watch(function () { ...
```

it should be:

```javascript
            __decorate([
                planck.Watch(function () { ...
```

It seems to work fine with class annotations.

**Comment by pankleks** [comment]
Any updates on that? are you able to reproduce? Can I help?

**Comment by mhegazy** [comment]
what version are you using? i do not seem to get it to repro on latest:

```js
var planck;
(function (planck) {
    function Watch(fn) {
        return function (target, key) { };
    }
    planck.Watch = Watch;
})(planck || (planck = {}));
var demo;
(function (demo) {
    var Test = (function () {
        function Test() {
        }
        __decorate([
            planck.Watch(function () { })
        ], Test.prototype, "Name");
        return Test;
    })();
})(demo || (demo = {}));
```

**Comment by pankleks** [comment]
I'm using latest official release - 1.6.2

I do have decorator function and class in two different files - maybe it has some influence.

**Comment by vladima** [comment]
interesting, I've tried to put `namespace planck` and `namespace demo` into two separate files and still cannot repro the issue. Also we have fixed a few similar issues in TypeScript 1.7 which is currently [available](http://blogs.msdn.com/b/typescript/archive/2015/11/30/announcing-typescript-1-7.aspx) so can you give it a try and see if you still experience the problem. If issue still exists it will be great if you can share a small repro so we can address the bug

**Comment by pankleks** [comment]
Installed 1.7 which seems to fix the issue.

Thanks a lot!
